client/job.go

```diff
@@ -374,6 +374,43 @@ func (q *JobClient) Create(jobPayload CreateJobPayload) (err error) {
 	return
 }
 
+// CreateJobInFolder creates a job in a specific folder and create folder first if the folder does not exist
+func (q *JobClient) CreateJobInFolder(jobPayload CreateJobPayload, path string) (err error) {
+	// check if folder exists
+	_, err = q.GetJob(path)
+	// create folder if not exist
+	if err != nil {
+		createFolderPayload := CreateJobPayload{
+			Name: path,
+			Mode: "com.cloudbees.hudson.plugins.folder.Folder",
+			From: jobPayload.From,
+		}
+		err = q.Create(createFolderPayload)
+		if err != nil {
+			return
+		}
+	}
```

Comment:
I prefer don't check and create the folder here. Users can check if the folder exists before calling this function.

Reply:
This function is quite similar to https://github.com/jenkins-zh/jenkins-cli/pull/610/files#diff-e829a497145e43bd7519ccab0bc1d6d68e11dc969faeec623b564216d3b1d13eR358.

Please consider reusing it. Such as, you can refactor the function `Create` to:

```
func (q *JobClient) Create(jobPayload CreateJobPayload) (err error) {
return CreateJobInFolder(jobPayload, "/")
}
```